Adding basic building guidelines into readme (#4)
davbrito/cpprex: README.md

# cpprex
A C++ implementation of Redux

## Building

### Building this library

You will need to have available a building stack including at least any compiler (gcc or msvc, preferably), cmake, and conan, and so, the software build essentials of your OS. Then just clone this repository and you can easily build with this command:

```sh
$ cd path/to/cpprex && \
  mkdir build && cd build && cmake .. -DCMAKE_BUILD_TYPE=Debug && make
```

**Note: this command was only ran on Bash at the time. It may vary according to your command line.**

**Note: release build type is not configured at the time yet.**

If everything goes fine, you might have these files in your build directory: `./bin/test_cpprex` and `./lib/libcpprex_.a`. These, asides the ones generated by cmake and conan. If so, you can test the build by running `ctest`. (On sources from stable branches should pass all tests.)

### Building a project with this library

> If you want to link this library to a project for usage, you may need to first build separately this project as the prior guidelines remarks. We recommend putting the library source code and binaries at `./deps/cpprex` relative directory of your application project source code directory. Make use of [Git submodules](https://gist.github.com/gitaarik/8735255) for integrate the source code of this repository into yours, and keep linked for future upgrades.

Reviewer: This is wrong!!!

CppRex is a header only library. As so it doesn't need to be built before usage.

Reviewer 2: Really? Great, then. I will fix that.

Reviewer 2: This was my redaction:

```suggestion
If you want to link this library to a project for its usage, at the time, we recommend putting the library source code at `./deps/cpprex` relative path of your application project directory. Make use of [Git submodules](https://gist.github.com/gitaarik/8735255) to integrate this repository into yours, but then, while keeping linked it for future upgrades.
```

I did some other changes about improving my English grammar in the other text.
